app/components/search_preview/search_preview.js

```diff
@@ -52,54 +58,89 @@ export default class SearchPreview extends PureComponent {
     };
 
     state = {
-        showPosts: false,
-        animationEnded: false
+        show: false,
+        error: false
     };
 
-    componentWillReceiveProps(nextProps) {
-        const {animationEnded, showPosts} = this.state;
-        if (animationEnded && !showPosts && nextProps.postIds.length) {
-            this.setState({showPosts: true});
+    constructor(props) {
+        super(props);
+
+        const {postIds} = props;
+        if (postIds && postIds.length >= 10) {
+            this.state.show = true;
```

Comment:
This looks kind of weird to do here since I'm not used to seeing direct manipulation of state. I'd lean towards just setting `this.state = ...` in the constructor, but I'm 0/5 really